src/Reader.mli

```diff
@@ -41,6 +41,8 @@ sig
 
   val run : env:env -> (unit -> 'a) -> 'a
   (** [run t] runs the thunk [t] which may perform reading effects. *)
+
+  val register_printer : ?read:string -> unit -> unit
```

Comment:
In this case, I think we should go with `register_printer read:string` because otherwise there's no need to use this function. Also, by "no documentation" I meant using the [stop command.](https://v2.ocaml.org/manual/ocamldoc.html#ss:ocamldoc-stop)

Reply:
Another possibility is to use `register_printer : read:(unit -> string option) -> unit` so that the users can plug in fancier functions.